paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -237,7 +237,7 @@ public class PaparazziPlugin @Inject constructor(
         test.inputs.dir(
           isVerifyRun.flatMap {
             project.objects.directoryProperty().apply {
-              set(if (it) snapshotOutputDir else null)
+              set(if (it && snapshotOutputDir.asFile.exists()) snapshotOutputDir else null)
```

Comment:
If you use a file collection instead you won't have this issue. This works but creating that directory will cause a configuration cache miss when the directory is later created I believe.

Reply:
yep, I noticed that in the gradle docs but didn't go that way. Let me check that

Reply:
Switching to this should work as a drop in replacement:
```kotlin
inputs.files(isVerifyRun.map { if (it) listOf(snapshotOutputDir) else emptyList() })
    .withPropertyName("paparazzi.snapshot.input.dir")
    .withPathSensitivity(PathSensitivity.RELATIVE)
    .optional()
```

Reply:
It works because calling `input.files` with an empty directory simply adds no input files

Reply:
To make caching work (and predictive test selection) the files themselves need to be included in the cache key so using a file property is not a good idea.

Reply:
yep makes sense, I opted to instead keep the directory property I originally had.

Reply:
The main issue with this approach is that the outcome can depend on when the task is configured. I also want to make clear that a directory property will still include the contents of the directory in the cache key for the task so you're still going to be checksumming all the snapshots (which is a good thing)